src/Core/Exporter.php

```diff
@@ -91,16 +103,19 @@ protected function getExportPath($base_path, $language)
     }
 
     /**
-     * Merge two arrays of translations and convert the resulting array to object.
-     * We don't override old strings to preserve existing translations.
+     * Merge two arrays of translations preserving existing translations and persistent strings.
      *
      * @param array $existing_strings
      * @param array $new_strings
-     * @return string
+     * @param array $persistent_strings
+     * @return array
      */
-    protected function mergeStrings($new_strings, $existing_strings)
+    protected function mergeStrings($new_strings, $existing_strings, $persistent_strings)
     {
-        return array_intersect_key(array_merge($new_strings, $existing_strings), $new_strings);
+        $merged_strings = array_merge($new_strings, $existing_strings);
+        return $this->arrayFilterByKey($merged_strings, function ($key) use ($persistent_strings, $new_strings) {
```

Comment:
@lasmanis I had to switch to a custom implementation of array_filter($array, $callback, ARRAY_FILTER_USE_KEY) to support PHP < 5.6.